frontend/src/app/services/book/book.service.ts

```diff
@@ -0,0 +1,70 @@
+import { Injectable } from '@angular/core';
+import { HttpClient, HttpErrorResponse, HttpHeaders } from '@angular/common/http';
+import { API_URL } from '../../config';
+import { catchError } from 'rxjs/internal/operators';
+import { Observable, throwError } from 'rxjs/index';
+
+export class Book {
+	id: number;
+	title: string;
+	authorName: string;
+	authorSurname: string;
+}
+
+
+@Injectable({
+	providedIn: 'root'
+})
+export class BookService {
+	url = API_URL + '/api/v1/library/books';
+	httpOptions = {
+		headers: new HttpHeaders({ 'Content-Type': 'application/json' })
+	};
+
+	constructor(private http: HttpClient) {
+	}
+
+	async getBooks(id?: number) {
+		let url;
+		if (!!id) {
+			url = this.url + `/${id}`;
+		} else {
+			url = this.url;
+		}
+		return await this.http.get<Book[]>(url).toPromise();
```

Comment:
IMO powinieneś zwracać listę książek zamiast 'Observable'. Między innymi dlatego że zrzucasz na komponent logikę związaną z obsługą błędów. I przede wszystkim to serwis powinien przechowywać stan aplikacji, czy też konkretnego wycinka stanu - w tym przypadku książek. Jeżeli pojawi się kolejny komponent który będzie potrzebował listy książek to znowu wykonasz strzał do BE, co kończy się tym że masz w dwóch miejscach takie same listy (a co gorsza te listy mogą się różnić między sobą i jest rozjazd). 
Przeczytaj to: https://blog.angular-university.io/how-to-build-angular2-apps-using-rxjs-observable-data-services-pitfalls-to-avoid/ szczególnie sekcja 'BehaviorSubject and how to use it'